app/screens/post_options/components/options/copy_permalink_option.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import Clipboard from '@react-native-community/clipboard';
+import React, {useCallback} from 'react';
+import {catchError, combineLatest, of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@constants/database';
+import {useServerUrl} from '@context/server';
+import {t} from '@i18n';
+import {dismissBottomSheet} from '@screens/navigation';
+import {WithDatabaseArgs} from '@typings/database/database';
+import SystemModel from '@typings/database/models/servers/system';
+import TeamModel from '@typings/database/models/servers/team';
+
+import BaseOption from './base_option';
+
+import type PostModel from '@typings/database/models/servers/post';
+
+const {SERVER: {SYSTEM, TEAM}} = MM_TABLES;
+const {CURRENT_TEAM_ID} = SYSTEM_IDENTIFIERS;
+
+type Props = {
+    teamName?: string;
+    post: PostModel;
+}
+const CopyPermalinkOption = ({teamName, post}: Props) => {
+    const serverUrl = useServerUrl();
+
+    const handleCopyLink = useCallback(() => {
+        const teamUrl = `${serverUrl}/${teamName}`;
```

Comment:
Potentially (in case of error)  `teamName` will be an empty string. Do we want to handle that?

Reply:
Refactored this part.